Use Hex diff library for diff and patch

dgoldie opened the pull request.

dgoldie committed:
Use Hex diff library for diff and patch

andyl commented:
Doug, thanks for this!

Please submit a diff solution that will convert to/from JSON.  The expression `Diff.diff("asdf", "assdf") |> Jason.encode!()` raises an error.

Also: if you can come up with a solution that does not require use of 3rd party dependency, that would be optimal.  But in the near term, I would for sure prefer a dependency vs shelling out to the linux utils.

FYI I added a bit more context to Issue #2. 

Thanks again for this PR!

andyl commented:
Just saw this on Elixir Forum - maybe a solution?

https://elixirforum.com/t/serialize-and-unserialize-structs-tuples-keyword-lists-atoms/40794

dgoldie committed:
diff and patch
- using Hex diff and patch
- convert diff to map to pass through json
- 3 test cases
  - insert row
  - delete row
  - insert and delete row
